### PR TITLE
Nav Mesh first tests

### PR DESCRIPTION
**Pull request by MadStark**

### CODE REVIEW

**Review by Xwilarg — CHANGES_REQUESTED** [review]
https://github.com/Unity-Technologies/NavMeshComponents

**Review by Xwilarg — DISMISSED** [review]

File: Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
+﻿using UnityEditor;
+using UnityEditor.AI;
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace Bug.Navigation
```
Comment (Xwilarg) [inline Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs]: I think the namespace sound be Bug.Editor

File: Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{
+	[CustomPropertyDrawer(typeof(AgentTypeAttribute))]
+	public class AgentTypeAttributeDrawer : PropertyDrawer
+	{
+		public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
+		{
+			int index = -1;
+			int count = NavMesh.GetSettingsCount();
+			string[] agentTypeNames = new string[count + 2];
```
Comment (Xwilarg) [inline Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs:16]: You can just write = new[count + 2]; here

**Review by Xwilarg — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs (hunk 2 above)
Comment (MadStark) [reply Assets/Scripts/Editor/AgentTypeAttributeDrawer.cs:16]: Thank you for the tip! I think I prefer it that way.

**Review by Xwilarg — APPROVED** [review]

**Pull request merged**